llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -2156,9 +2164,13 @@ bool SimplifyCFGOpt::hoistSuccIdenticalTerminatorToSwitchOrIf(
       Updates.push_back({DominatorTree::Insert, TIParent, Succ});
   }
 
-  if (DTU)
-    for (BasicBlock *Succ : successors(TI))
+  if (DTU) {
+    // TI might be a switch with multi-cases destination, so we need to care for
+    // the duplication of successors.
+    SmallPtrSet<BasicBlock *, 4> UniqSuccs(llvm::from_range, successors(TI));
```

Comment:
Why can't we use the existing UniqSuccessors here and need to compute it again?

Reply:
There are not always too many successors such that recalculating the uniq successors in-place is acceptable.

If reusing `UniqSuccessors` here, we need to add a new parameter to the function only for this potential usage (if DTU is nullptr, this parameter is useless).